crates/forge/src/main.rs

```diff
@@ -152,6 +157,24 @@ fn combine_configs(
     )
 }
 
+fn write_failed_tests(all_failed_tests: &[TestCaseSummary]) -> Result<()> {
+    let test_fails = std::env::current_dir()?.join(".prev_tests_failed");
```

Comment:
This doesn't seem safe. We should be using a workspace root from scarb to resolve directory for storing `.prev_tests_failed`. You can take a look at this: https://github.com/foundry-rs/starknet-foundry/blob/fb18cdd9ae335677ea12a2448dc86533d3df2052/crates/forge/src/running.rs#L353

Reply:
After that, we should write/extend tests using scarb workspaces so we can verify that this approach works with workspaces.